huaweicloud/services/dew/resource_huaweicloud_kps_keypair.go

```diff
@@ -343,3 +338,60 @@ func updateDesc(client *kps.KpsClient, id, desc string) diag.Diagnostics {
 
 	return nil
 }
+
+func updatePrivateKey(client *kps.KpsClient, d *schema.ResourceData) diag.Diagnostics {
+	privateKey := d.Get("private_key").(string)
+	// clear kps keypair privateKey
+	if privateKey == "" {
+		clearOps := &model.ClearPrivateKeyRequest{
+			KeypairName: d.Get("name").(string),
+		}
+		_, err := client.ClearPrivateKey(clearOps)
+		if err != nil {
+			return diag.Errorf("error deleting KPS keypair privateKey: %s", err)
+		}
+	}
+
+	// import kps keypair privateKey
+	if privateKey != "" {
+		importOps, err := buildImportPrivateKeyParams(d)
+		if err != nil {
+			diag.Errorf("error building KPS keypair import privateKey params: %s", err)
+		}
+		_, importError := client.ImportPrivateKey(importOps)
+		if importError != nil {
+			return diag.Errorf("error importing KPS keypair privateKey: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func buildImportPrivateKeyParams(d *schema.ResourceData) (*model.ImportPrivateKeyRequest, error) {
+	importOps := &model.ImportPrivateKeyRequest{
+		Body: &model.ImportPrivateKeyRequestBody{
+			Keypair: &model.ImportPrivateKeyKeypairBean{
+				Name: d.Get("name").(string),
+			},
+		},
+	}
+
+	t := d.Get("encryption_type").(string)
```

Comment:
if t == "" {
  return nil, fmt.Errorf("field `encryption_type` is required when import a private key.")
}